File: Data Structures & Algorithms/subsets-ii/submission-0.java
class Solution {
    List<List<Integer>> result =  new ArrayList<>();
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums); 
        backtrack(nums,0,new ArrayList<>());
        result.add(new ArrayList<>());
        return result;   
    }
    public void backtrack(int[] nums,int start,List<Integer> path){
        if(nums.length==path.size())
        {
            return;
        }
        for(int i=start;i<nums.length;i++)
        {   
            if (i > start && nums[i] == nums[i - 1]) continue;
            path.add(nums[i]);
            result.add(new ArrayList<>(path));
            backtrack(nums,i+1,path);
            path.remove(path.size()-1);
        }
        
    }
}

File: Data Structures & Algorithms/decode-string/submission-0.cpp
class Solution {
public:
    int i = 0;

    string decodeString(string s) {
        string result = "";
        int k = 0;

        while (i < s.length()) {
            char ch = s[i];

            if (isdigit(ch)) {
                k = k * 10 + (ch - '0');
                i++;
            }
            else if (ch == '[') {
                i++;

                string decodedStr = decodeString(s);

                for (int j = 0; j < k; j++) {
                    result += decodedStr;
                }

                k = 0;
            }
            else if (ch == ']') {
                i++;
                break;
            }
            else {
                result += ch;
                i++;
            }
        }

        return result;
    }
};